### PR TITLE
Free actions

### PR DESCRIPTION
**Pull request by rzulian**
This is implementing standard free actions. I decided to rename bowls to area, accordingly to GP rules.
I've created resource management for each different power area.
I didn't managed the free action spend a QIC to extend range, as this should be included in the distance calculation.

### CODE REVIEW

**Comment by rzulian** [comment]
There are two different Discard actions: discard from the game (e.g federation for satellites, from area2 to area3), and discard to Gaia. So we need:
GainToken ( -GainToken can be Discard outside);
BurnPower ( from area2 to area3)
DiscardGaia
ChargePower( -ChargePower is Spend)

**Comment by coyotte508** [comment]
You're right. `DiscardGaia` can be `MoveToGaia`. I don't know which acronym to use for this resource? `mv`? `pw2g`? `g`? `gaia`?

**Comment by coyotte508** [comment]
I'm don't think `BurnPower` needs to be added as a resource, since it has its own command

**Comment by rzulian** [comment]
Can you solve those conflicts on your side, or what should I do?

**Comment by coyotte508** [comment]
I'm finishing up on something and going to sleep, I'll look at it tomorrow evening

**Comment by rzulian** [comment]
Ok. I'll start working on power actions

**Comment by rzulian** [comment]
I figured it out, wrong merge in imports :(

File: src/player-data.ts [hunk 1]
```diff
@@ -142,60 +143,58 @@ export default class PlayerData extends EventEmitter {
...
-    power -= bowl1ToGaia;
+  spendPower(power: number)  {  
+      this.power.area3 -= power;
+      this.power.area1 += power;
+  }
 
-    if (power <= 0) {
-      return;
-    }
+  discartPower(power: number, type: Resource) {
```
Comment (coyotte508) [inline src/player-data.ts]: discard, not discart

File: src/map.ts [hunk 2]
```diff
@@ -128,7 +128,7 @@ export default class SpaceMap {
   }
 
   excludedHexesForBuildingFederation(player: Player) {
-    const ret: Set<GaiaHex> = new Set();
+    const ret: Set<Hex<GaiaHexData>> = new Set();
```
Comment (coyotte508) [inline src/map.ts]: change can be removed

File: src/sector.ts [hunk 3]
```diff
@@ -1,5 +1,15 @@
 import {Hex, Grid} from "hexagrid";
 import { Planet, Building, Player } from "./enums";
+
+export interface GaiaHexData {
```
Comment (coyotte508) [inline src/sector.ts]: all this can be removed

File: src/player-data.ts (hunk 1 above)
Comment (rzulian) [reply src/player-data.ts]: oops

File: src/sector.ts (hunk 3 above)
Comment (rzulian) [reply src/sector.ts]: ok

**Pull request merged**